hi-ml-cpath/src/health_cpath/utils/plots_utils.py

```diff
@@ -216,13 +219,17 @@ def __init__(
         self.should_upscale_coords = loading_params.should_upscale_coordinates()
         self.loading_params.set_roi_type_to_foreground()
         self.slides_dataset: Optional[SlidesDataset] = None
+        self.ihc_slides_dataset: Optional[SlidesDataset] = None
 
-    def get_slide_dict(self, slide_node: SlideNode) -> SlideDictType:
-        """Returns the slide dictionary for a given slide node"""
-        assert self.slides_dataset is not None, "Cannot plot attention heatmap or wsi without slides dataset"
-        slide_index = self.slides_dataset.dataset_df.index.get_loc(slide_node.slide_id)
+    def get_slide_dict(self, slide_node: SlideNode, slides_dataset: SlidesDataset) -> SlideDictType:
+        """Returns the slide dictionary for a given slide node from a slides dataset.
+
+        :param slide_node: The slide node that encapsulates the slide metadata.
+        :param slides_dataset: The slides dataset that contains the slide image and other metadata.
+        """
+        slide_index = slides_dataset.dataset_df.index.get_loc(slide_node.slide_id)
```

Comment:
What if the slide_id is not found? would be good to have an error message that also prints out the missing slide_id.

Reply:
Very good point, my jobs all failed because of a missing slide, thanks for catching it